vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt

```diff
@@ -239,6 +239,20 @@ class VectorPreferences @Inject constructor(private val context: Context) {
     }
 
     private val defaultPrefs = DefaultSharedPreferences.getInstance(context)
+    
+    /**
+     * Allow subscribing and unsubscribing to configuration changes. This is
+     * particularly useful when you need to be notified of a configuration change
+     * in a background service, e.g. for the P2P demos.
+     */
+    
+    public fun subscribeToChanges(listener: SharedPreferences.OnSharedPreferenceChangeListener) {
```

Comment:
`import android.content.SharedPreferences` is missing

Also `public` modifier can be removed, this is the default value in Kotlin